Improve error when using the dependencies kwarg

nirbheek opened the pull request:
The error message is misleading (talks about external dependencies), and doesn't tell you what you need to do (use the output of `declare_dependency`, `dependency`, or `find_library`). At the same time rename `add_external_deps` to `add_deps` since it adds internal deps too.

Plus many more error message improvements all over the place.

jpakkane commented:
Those long comments should be multiline strings. Mixing split strings with `\n` is close to unreadable.

QuLogic commented on mesonbuild/build.py:
```diff
@@ -626,10 +626,13 @@ def add_external_deps(self, deps):
-You probably wanted to use a dependency declared in it instead. Access it
-by calling get_variable() on the subproject object.''')
-                raise InvalidArguments('Argument is not an external dependency.')
+                    raise InvalidArguments('Tried to use subproject object as a dependency. ' \
```
This is not equivalent since the former included the newlines and this one doesn't. Also, the backslash at the end is not necessary here.

jpakkane commented:
Conflicts ...

nirbheek commented:
Rebased and pushed; also fixed the newline issue that @QuLogic raised.

I don't like multiline strings because they completely destroy the indentation of the function, but I'll convert it to use that if you have a strong opinion regarding that.

jpakkane commented:
Conflicts again...

nirbheek commented:
Fixed again!
